jujugui/static/gui/src/app/app.js

```diff
@@ -1698,28 +1658,86 @@ YUI.add('juju-gui', function(Y) {
 
     /**
       Handle the request to display the new model state.
-
-      @method _handleNewModel
     */
     _handleNewModel: function() {
       this.switchEnv();
     },
 
     /**
-      Handle the request to display the user entity state.
-
-      @method _handleUserEntity
+      Handles the state changes for the model key.
       @param {Object} state - The application state.
       @param {Function} next - Run the next route handler, if any.
     */
-    _handleUserEntity: function(state, next) {
-      const userPath = state.user;
-      const handleEntity = this._handleUserEntityPath.bind(
-        this, state, next, userPath);
-      // Attempt to handle an exisitng path.
-      if (!handleEntity()) {
-        // If the path has not been visited then populate the path storage with
-        // the list of models.
+    _handleModelState: function(state, next) {
+      if (this.get('modelUUID') !== state.model.uuid) {
+        this._switchModelToUUID(state.model.uuid);
+      }
+      next();
+    },
+
+    /**
+      Switches to the specified UUID, or if none is provided then
+      switches to the unconnected mode.
+      @param {String} uuid The uuid of the model to switch to, or none.
+    */
+    _switchModelToUUID: function(uuid) {
+      let socketURL = undefined;
+      if (uuid) {
+        this.set('modelUUID', uuid);
+        socketURL = this.createSocketURL(this.get('socketTemplate'), uuid);
+      } else {
+        this.set('modelUUID', undefined);
+        console.log('no uuid or model name defined: using unconnected mode');
+      }
+      this.switchEnv(socketURL);
+    },
+
+    /**
+      Determines if the user state is a store path or a model path.
+      @param {String} The state value for the 'user' key.
+      @return {Promise} A promise with the charmstore entity if one exists.
+    */
+    _fetchEntityFromUserState: function(userState) {
+      const urlParts = window.jujulib.URL.fromString('u/' + userState);
+      const URLlib = new window.jujulib.URL(urlParts);
```

Comment:
This is not required, as `window.jujulib.URL.fromString` already returns a URL.
You can just do:
```javascript
const legacyPath = window.jujulib.URL.fromString('u/' + userState).legacyPath();
```
and that's it. Or, alternatively, the URLlib name is weird, so
```javascript
const url = window.jujulib.URL.fromString('u/' + userState);
const legacyPath = url.legacyPath();
```